src/plugins/intel_gpu/include/intel_gpu/primitives/custom_gpu_primitive.hpp

```diff
@@ -65,15 +120,23 @@ struct custom_gpu_primitive : public primitive_base<custom_gpu_primitive> {
                          const std::string& build_options,
                          const layout& output_layout,
                          const std::vector<size_t>& gws = {},
-                         const std::vector<size_t>& lws = {})
+                         const std::vector<size_t>& lws = {},
+                         const std::shared_ptr<ov::Node>& op = nullptr,
+                         const int iidx = -1,
+                         const std::vector<std::string> globalSizeRules = {},
+                         const std::vector<std::string> localSizeRules = {})
         : primitive_base(id, inputs, 1, {optional_data_type()}, {output_layout.data_padding}),
           kernel_entry_point(kernel_entry_point),
           kernel_arguments(kernel_arguments),
           build_options(build_options),
           output_layout(output_layout),
           gws(gws.size() ? gws : std::vector<size_t>{output_layout.count()}),
           lws(lws),
-          kernels_code(kernels_code) {}
+          kernels_code(kernels_code),
+          op(op),
```

Comment:
Maybe better to clone it? Not sure if there are some ops references kept, the model could be released to recycle the memory.

Reply:
updated in https://github.com/openvinotoolkit/openvino/pull/30880/files#diff-a061b37e01611c75206b2022772380b257eb76b31615a178baddffe33a1eb404R208-R214